lib/omniauth/strategies/saml.rb

```diff
@@ -103,15 +109,23 @@ def other_phase
       uid { @name_id }
 
       info do
-        {
-          :name  => @attributes[:name],
-          :email => @attributes[:email] || @attributes[:mail],
-          :first_name => @attributes[:first_name] || @attributes[:firstname] || @attributes[:firstName],
-          :last_name => @attributes[:last_name] || @attributes[:lastname] || @attributes[:lastName]
-        }
+        found_attributes = options.attribute_statements.map do |key, values|
+          attribute = find_attribute_by(values)
+          [key, attribute]
+        end
+
+        found_attributes.to_h
```

Comment:
Array#to_h is not present in Ruby 2.0 and earlier.  Using `Hash[found_attributes]` instead seems to work under Ruby 2.0 and is still supported under Ruby 1.9.3 and 2.2 according to the docs.